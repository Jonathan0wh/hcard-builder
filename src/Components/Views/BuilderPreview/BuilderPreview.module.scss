@import 'styles/colors';

.container {
  background-color: $background-grey;
  height: 100%;
  overflow: scroll;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  :global {
    .vcard {
      background-color: $white;
      box-shadow: 0 2px 2px $shadow-grey;
      width: 375px;

      @media only screen and (min-width: 992px) {
        width: 430px;
        height: 306px;
      }
    }
    .fn {
      display: inline-block;
      margin-top: 55px;
      margin-left: 25px;
    }
    .photo {
      float: right;
      width: 87px;
      height: 108px;
      margin-top: 10px;
      margin-right: 12px;
      object-fit: cover;
      box-shadow: 0 2px 2px $shadow-grey;
    }
    .postal-code {
      display: inline-block;
      width: 90px;

      @media only screen and (min-width: 992px) {
        width: 120px;
      }
    }
  }
}

.preview_title_container {
  @media only screen and (min-width: 992px) {
    width: 430px;
    text-align: right;
  }
}

.preview_title {
  font-size: 16px;
  font-weight: 400;
  color: $text-grey;
}

.preview_identity {
  height: 100px;
  background-color: $bluewood;
  color: $white;
  font-family: 'Merriweather';
  font-size: 24px;
  font-weight: 700;
}

.preview_detail {
  padding: 30px 25px 26px 25px;
}

.preview_row {
  border-bottom-color: $divider-grey;
  border-bottom-width: 1px;
  border-bottom-style: solid;
  margin-bottom: 6px;
  height: 24px;
  white-space: nowrap;
}

.preview_row_additional {
  padding-left: 75px;
}

.preview_label {
  font-size: 10px;
  font-weight: 400;
  color: $text-bluewood;
  display: inline-block;
  width: 75px;
}

.preview_label_country {
  width: 60px;
}

.preview_content {
  font-family: 'Merriweather';
  font-size: 16px;
  font-weight: 400;
  color: $black;
}
