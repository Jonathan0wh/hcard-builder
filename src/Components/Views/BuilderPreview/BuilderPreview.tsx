import React from 'react';
import { Typography } from 'antd';
import classNames from 'classnames';

import { IFormData } from 'Components/Views/HCardBuilder';
import { STRINGS_PREVIEW } from 'configs/strings';
import defaultAvatar from 'assets/images/avatar_default.png';
import styles from './BuilderPreview.module.scss';

interface BuilderPreviewProps {
  formData: IFormData;
}

const BuilderPreview = ({ formData }: BuilderPreviewProps) => {
  const { Text } = Typography;

  return (
    <div className={styles.container}>
      <div className={styles.preview_title_container}>
        <Text className={styles.preview_title}>
          {STRINGS_PREVIEW.PREVIEW_TITLE}
        </Text>
      </div>

      <div className="vcard">
        <div className={styles.preview_identity}>
          <img
            src={
              formData.avatar
                ? URL.createObjectURL(formData.avatar)
                : defaultAvatar
            }
            alt="avatar"
            className="photo"
          />
          <span className="fn">
            <span className="given-name">
              {formData.givenName} {formData.surname}
            </span>
            <span className="family-name"></span>
          </span>
        </div>

        <div className={styles.preview_detail}>
          <div className={styles.preview_row}>
            <span className={styles.preview_label}>
              {STRINGS_PREVIEW.PREVIEW_LABEL_EMAIL}
            </span>
            <span className={classNames('email', styles.preview_content)}>
              {formData.email}
            </span>
          </div>
          <div className={styles.preview_row}>
            <span className={styles.preview_label}>
              {STRINGS_PREVIEW.PREVIEW_LABEL_PHONE}
            </span>
            <span className={classNames('tel', styles.preview_content)}>
              {formData.tel}
            </span>
          </div>
          <div className="adr">
            <div className={styles.preview_row}>
              <span className={styles.preview_label}>
                {STRINGS_PREVIEW.PREVIEW_LABEL_ADDRESS}
              </span>
              <span
                className={classNames('street-address', styles.preview_content)}
              >
                {`${formData.house}${formData.house && ' '}${formData.street}`}
              </span>
            </div>
            <div
              className={classNames(
                styles.preview_row,
                styles.preview_row_additional
              )}
            >
              <span className={classNames('locality', styles.preview_content)}>
                {`${formData.suburb}${formData.suburb &&
                  formData.state &&
                  ', '}`}
              </span>
              <span className={classNames('region', styles.preview_content)}>
                {formData.state}
              </span>
            </div>
          </div>
          <div className={styles.preview_row}>
            <span>
              <span className={styles.preview_label}>
                {STRINGS_PREVIEW.PREVIEW_LABEL_POSTCODE}
              </span>
              <span
                className={classNames('postal-code', styles.preview_content)}
              >
                {formData.postcode}
              </span>
            </span>
            <span>
              <span
                className={classNames(
                  styles.preview_label,
                  styles.preview_label_country
                )}
              >
                {STRINGS_PREVIEW.PREVIEW_LABEL_COUNTRY}
              </span>
              <span
                className={classNames('country-name', styles.preview_content)}
              >
                {formData.country}
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  );
};

export default BuilderPreview;
