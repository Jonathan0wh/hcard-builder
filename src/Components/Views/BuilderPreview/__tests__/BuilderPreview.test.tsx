import React from 'react';
import { shallow } from 'enzyme';

import BuilderPreview from '../BuilderPreview';

describe('<BuilderPreview />', () => {
  const givenName = 'Sam',
    surname = 'Fairtax',
    email = 'sam.fairtax@fairtaxmedia.com.au',
    tel = '02 9282 2833',
    house = '1',
    street = 'Darling Island Road',
    suburb = 'Pyrmont',
    state = 'NSW',
    postcode = '2009',
    country = 'Australia',
    avatar = null;

  const formData = {
    givenName,
    surname,
    email,
    tel,
    house,
    street,
    suburb,
    state,
    postcode,
    country,
    avatar
  };

  it('should render by default', () => {
    const wrapper = shallow(<BuilderPreview formData={formData} />);
    expect(wrapper).toMatchSnapshot();
  });
});
