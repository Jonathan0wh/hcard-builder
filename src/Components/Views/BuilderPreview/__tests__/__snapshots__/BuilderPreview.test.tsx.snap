// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BuilderPreview /> should render by default 1`] = `
<div
  className="container"
>
  <div
    className="preview_title_container"
  >
    <Text
      className="preview_title"
    >
      HCARD PREVIEW
    </Text>
  </div>
  <div
    className="vcard"
  >
    <div
      className="preview_identity"
    >
      <img
        alt="avatar"
        className="photo"
        src="avatar_default.png"
      />
      <span
        className="fn"
      >
        <span
          className="given-name"
        >
          Sam
           
          Fairtax
        </span>
        <span
          className="family-name"
        />
      </span>
    </div>
    <div
      className="preview_detail"
    >
      <div
        className="preview_row"
      >
        <span
          className="preview_label"
        >
          EMAIL
        </span>
        <span
          className="email preview_content"
        >
          sam.fairtax@fairtaxmedia.com.au
        </span>
      </div>
      <div
        className="preview_row"
      >
        <span
          className="preview_label"
        >
          PHONE
        </span>
        <span
          className="tel preview_content"
        >
          02 9282 2833
        </span>
      </div>
      <div
        className="adr"
      >
        <div
          className="preview_row"
        >
          <span
            className="preview_label"
          >
            ADDRESS
          </span>
          <span
            className="street-address preview_content"
          >
            1 Darling Island Road
          </span>
        </div>
        <div
          className="preview_row preview_row_additional"
        >
          <span
            className="locality preview_content"
          >
            Pyrmont, 
          </span>
          <span
            className="region preview_content"
          >
            NSW
          </span>
        </div>
      </div>
      <div
        className="preview_row"
      >
        <span>
          <span
            className="preview_label"
          >
            POSTCODE
          </span>
          <span
            className="postal-code preview_content"
          >
            2009
          </span>
        </span>
        <span>
          <span
            className="preview_label preview_label_country"
          >
            COUNTRY
          </span>
          <span
            className="country-name preview_content"
          >
            Australia
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
